pkg/proc/gdbserial/gdbserver.go

```diff
@@ -1126,13 +1126,36 @@ func (p *gdbProcess) restartWorker(cctx *proc.ContinueOnceContext, pos string) (
 	}
 
 	// Is this a checkpoint on a server using local checkpoints?
-	if len(pos) > 1 && pos[:1] == "c" && p.conn.isUndoServer {
-		cpid, _ := strconv.Atoi(pos[1:])
-		checkpoint, exists := p.localCheckpoints[cpid]
-		if !exists {
-			return nil, errors.New("Checkpoint not found")
+	if p.conn.isUndoServer {
+		// Validate and transform input.
+		if pos == "start" || pos == "end" {
+			// Special case values - valid with no extra checking.
+		} else if len(pos) > 1 && pos[:1] == "c" {
+			// Validate a checkpoint ID.
+			cpid, _ := strconv.Atoi(pos[1:])
+			checkpoint, exists := p.localCheckpoints[cpid]
+			if !exists {
+				return nil, errors.New("Checkpoint not found")
+			}
+			pos = checkpoint.When
+		} else {
+			// Validate a potential bbcount or precise time.
+			pos = strings.ReplaceAll(pos, ",", "")
+			var bbcount, pc uint64
+			var err error
+			if strings.Contains(pos, ":") {
+				_, err = fmt.Sscanf(pos, "%d:0x%x\n", &bbcount, &pc)
+			} else if _, err = fmt.Sscanf(pos, "%d\n", &bbcount); err == nil {
+				// It's a valid bbcount.
+				pc = 0
+			}
+
+			if err != nil {
+				return nil, errors.New("Could not parse time or checkpoint argument to restart.")
+			}
+
+			pos = fmt.Sprintf("%x;%x", bbcount, pc)
```

Comment:
I think a comment explaining the format would be helpful here.

Reply:
Format of the output `pos` or of the incoming argument? (or both?)